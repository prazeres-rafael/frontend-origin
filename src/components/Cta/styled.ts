import styled from 'styled-components';
import { Button } from '@mui/material';
import { fontFamily } from 'utils/font';
import { palette } from 'utils/palette';

export const ButtonContent = styled.span({
  margin: '18px 128px',
  fontFamily: fontFamily,
  textTransform: 'none',
  fontSize: 16,
  fontWeight: 600,
});

export const BaseButton = styled(Button)({
  backgroundColor: palette.brand.primary,
  borderRadius: 32,
});
