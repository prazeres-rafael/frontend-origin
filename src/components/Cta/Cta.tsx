import React from 'react';
import { BaseButton, ButtonContent } from './styled';

type Props = {
  children: React.ReactNode;
};

export default function Cta({ children }: Props): React.ReactElement {
  return (
    <BaseButton variant="contained">
      <ButtonContent>{children}</ButtonContent>
    </BaseButton>
  );
}
