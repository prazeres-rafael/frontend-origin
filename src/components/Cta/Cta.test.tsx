import { render } from '@testing-library/react';
import React from 'react';

import Cta from './Cta';

describe('Cta', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Cta>Test cta component</Cta>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
