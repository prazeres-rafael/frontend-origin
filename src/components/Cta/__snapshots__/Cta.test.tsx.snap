// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cta should match the snapshot 1`] = `
<button
  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root styled__BaseButton-sc-yzlx9g-1 fKkJnT css-sghohy-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  <span
    class="styled__ButtonContent-sc-yzlx9g-0 kHHqy"
  >
    Test cta component
  </span>
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
